pkg/storage/workflow_client.go

```diff
@@ -0,0 +1,182 @@
+func (n workflowClient) DownloadArtifactStream(ctx context.Context, req *pb.DownloadArtifactRequest, onChunk func(*pb.DownloadArtifactChunk) error) error {
+	if n.jwtGenerator != nil {
+		var err error
+		ctx, err = n.injectToken(ctx, req)
+		if err != nil {
+			return err
+		}
+	}
+	n.log.Infow("DownloadArtifactStream RPC called", "id", req.Id, "type", req.Type.String(), "environment", req.Environment.String())
+
+	stream, err := n.client.DownloadArtifact(ctx, req)
+	if err != nil {
+		n.log.Errorw("Failed to initiate artifact download stream", "id", req.Id, "error", err)
+		return err
+	}
+
+	for {
+		chunk, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			n.log.Errorw("Error while receiving artifact chunk", "id", req.Id, "error", err)
+			return err
+		}
+		if err := onChunk(chunk); err != nil {
+			n.log.Errorw("Error in chunk callback", "id", req.Id, "error", err)
+			return err
+		}
+	}
+
+	n.log.Infow("Successfully streamed artifact", "id", req.Id)
+	return nil
+}
+func (n workflowClient) Close() error {
+	err := n.conn.Close()
+	if err != nil {
+		n.log.Errorw("Failed to close WorkflowClient connection", "error", err)
+		return err
+	}
+	n.log.Infow("Closed WorkflowClient connection")
+	return nil
+}
+
+// NodeClientOpt is a functional option type for configuring the WorkflowClient
+type NodeClientOpt func(*nodeConfig)
+
+type nodeConfig struct {
+	log                  logger.Logger
+	transportCredentials credentials.TransportCredentials
+	jwtGenerator         *nodeauth.NodeJWTGenerator // Optional JWT manager for authentication
+}
+
+// defaultNodeConfig returns a default configuration for the WorkflowClient
+func defaultNodeConfig() nodeConfig {
+	loggerInst, _ := logger.New()
+	return nodeConfig{
+		log:                  loggerInst,
+		transportCredentials: credentials.NewTLS(nil), // Use default TLS credentials
+		jwtGenerator:         nil,                     // No JWT generator by default
+	}
+}
+
+// WithLogger sets the logger for the WorkflowClient
+func WithLogger(log logger.Logger) NodeClientOpt {
+	return func(cfg *nodeConfig) {
+		cfg.log = log
+	}
+}
+
+// WithTransportCredentials sets the transport credentials for the WorkflowClient
+func WithTransportCredentials(creds credentials.TransportCredentials) NodeClientOpt {
+	return func(cfg *nodeConfig) {
+		cfg.transportCredentials = creds
+	}
+}
+
+func WithJWTGenerator(jwtGenerator *nodeauth.NodeJWTGenerator) NodeClientOpt {
+	return func(cfg *nodeConfig) {
+		cfg.jwtGenerator = jwtGenerator
+	}
+}
+
+// NewNodeClient creates a new WorkflowClient with the specified address and options
+// It returns a WorkflowClient which can DownloadArtifacts
+func NewNodeClient(ctx context.Context, address string, opts ...NodeClientOpt) (WorkflowClient, error) {
+	cfg := defaultNodeConfig()
+
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	grpcOpts := []grpc.DialOption{
+		grpc.WithTransportCredentials(cfg.transportCredentials),
```

Comment:
For this to work in local development environments on a localhost URL, it is going to need insecure credentials.
See billing client [here](https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/billing/workflow_client.go#L58).